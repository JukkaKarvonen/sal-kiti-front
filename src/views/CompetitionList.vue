<template>
  <div class="event">
    <Event v-on:update:createPermission="createPermission = $event" />
    <CompetitionList
      v-bind:createPermission="createPermission"
    ></CompetitionList>
  </div>
</template>

<script>
/**
 * Competition list view
 */

import CompetitionList from "@/components/CompetitionList.vue";
import Event from "@/components/EventInformation.vue";

export default {
  name: "CompetitionListView",
  data() {
    return {
      createPermission: false
    };
  },
  components: {
    Event,
    CompetitionList
  },
  created() {
    document.title = this.$t("title.event") + " | " + this.$t("title.prefix");
  }
};
</script>
