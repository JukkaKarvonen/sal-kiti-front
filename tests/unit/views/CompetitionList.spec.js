import { shallowMount, createLocalVue } from "@vue/test-utils";
import CompetitionList from "@/views/CompetitionList.vue";

import BootstrapVue from "bootstrap-vue";

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe("CompetitionList.vue", () => {
  const $route = {
    params: { id: 1 }
  };
  it("renders content", () => {
    const wrapper = shallowMount(CompetitionList, {
      localVue,
      mocks: {
        $route
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
