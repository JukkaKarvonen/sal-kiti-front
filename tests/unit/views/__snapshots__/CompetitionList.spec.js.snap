// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompetitionList.vue renders content 1`] = `
<div
  class="event"
>
  <event-stub />
   
  <competitionlist-stub />
</div>
`;
